### PR TITLE
Evaluate on start

### PR DESCRIPTION
**Pull request by dl1jbe**
This is a first part to fix issue #248.

Besides some cleanup, simplification and restructuring it changes addcall() to accept a qso_t record of relevant 
information. That gets collected by a new function collect_qso_data() from the bunch of global variables.
Long term goal is to use the same addcall() function for evaluation of newly made QSOs and of contacts red from the log file during readcalls().

Furthermore the patch uses parse_qso() to get these information from QSOs coming in from LAN (in addcall2() ) or from reading the already written log file (in readcalls() ).

Next steps will be the use of the qso_t record for addmult() and maybe makelogline(). Furthermore the parsed qso_t struct and the evaluation results for all qsos should be kept in memory instead( or in complement) of the actual qsos[] array which stores only the plain log lines and needs a reevalution every time some value is needed from them.

### CODE REVIEW

**Comment by dl1jbe** [comment]
There are some severe changes. Could use a thorough review.

File: src/addcall.c [hunk 1]
```diff
@@ -112,96 +163,53 @@ int addcall(void) {
...
-			    == 0) {
-			pfxnummulti[pfxnumcntidx].qsos[pxnr] |= inxes[bandinx];
-			addcallarea = 1;
-			countryscore[bandinx]++;
-			zonescore[bandinx]++;
-		    }
-		}
-		break;
 
+    if (add_ok == 1) {
```
Comment (zcsahok) [inline src/addcall.c:191]: `add_ok` could be a boolean

File: src/addcall.c [hunk 2]
```diff
@@ -290,70 +287,39 @@ int addcall2(void) {
...
-			zones[zone] |= inxes[bandinx];
-		    }
-		    break;
-
+	    if (pfxnumcntidx < 0) {
+		if (cty != 0 && (countries[cty] & inxes[bandinx]) == 0) {
+		    countries[cty] |= inxes[bandinx];
+		    countryscore[bandinx]++;
+//                  new_cty = cty;
+	    }
```
Comment (zcsahok) [inline src/addcall.c:310]: strange indent, maybe astyle is need

File: src/readcalls.c [hunk 3]
```diff
@@ -240,20 +203,22 @@ int readcalls(void) {
...
+	update_worked(station, qso);
 
-	countrynr = getctydata(presentcall);
 
 	if (continentlist_only) {
-	    if (!is_in_continentlist(continent)) {
+	    if (!is_in_continentlist(dxcc_by_index(countrynr)->continent)) {
 		qsos_per_band[bandindex]++;
+		free_qso(qso);
+		qso = NULL;
```
Comment (zcsahok) [inline src/readcalls.c:221]: is this needed? or just a preventive measure in case `qso` would be used later on?

File: src/tlf.h [hunk 4]
```diff
@@ -189,12 +190,14 @@ typedef struct {
 struct qso_t {
     char *logline;
     int band;
+    int bandindex;
```
Comment (zcsahok) [inline src/tlf.h:193]: (can't comment on unchanged line)
```char exchange[13];```  should be 24 ((77-55+1)+1) to be able to store a QSO-mode exchange

**Review by zcsahok — APPROVED** [review]
Found mostly just minor stuff. Thanks.

File: src/addcall.c (hunk 1 above)
Comment (dl1jbe) [reply src/addcall.c]: Thanks. Just fixed.

File: src/addcall.c (hunk 2 above)
Comment (dl1jbe) [reply src/addcall.c:310]: Good catch. Vi's autoindent had difficulties with the commented out line above.

File: src/readcalls.c (hunk 3 above)
Comment (dl1jbe) [reply src/readcalls.c:221]: Not really. But it helps to catch 'use after free' errors during development.

File: src/tlf.h (hunk 4 above)
Comment (dl1jbe) [reply src/tlf.h:193]: You mean in declaration of worked_t? You are right - corrected.

**Review by zcsahok — APPROVED** [review]
Thanks, will merge into dev.

**Pull request merged**